mlrun/feature_store/retrieval/local_merger.py

```diff
@@ -116,6 +178,8 @@ def _asof_join(
             left_on=entity_timestamp_column,
             right_on=featureset.spec.timestamp_key,
             by=indexes,
+            left_by=left_keys,
+            right_by=right_keys,
```

Comment:
To make sure I understand - `outer` join is not really possible when doing asof-join, right? So this needs to be documented or a warning needs to be issued or something similar.